store/sqlstore/group_supplier.go

```diff
@@ -0,0 +1,864 @@
+	}
+
+	groupSyncables := []*model.GroupSyncable{}
+
+	switch syncableType {
+	case model.GroupSyncableTypeTeam:
+		sqlQuery := `
+			SELECT
+				GroupTeams.*, 
+				Teams.DisplayName AS TeamDisplayName, 
+				Teams.Type AS TeamType
+			FROM 
+				GroupTeams
+				JOIN Teams ON Teams.Id = GroupTeams.TeamId
+			WHERE 
+				GroupId = :GroupId AND GroupTeams.DeleteAt = 0`
+
+		results := []*groupTeamJoin{}
+		_, err := s.GetMaster().Select(&results, sqlQuery, args)
+		if err != nil {
+			result.Err = appErrF(err.Error())
+			return result
+		}
+		for _, result := range results {
+			groupSyncable := &model.GroupSyncable{
+				SyncableId:      result.TeamId,
+				GroupId:         result.GroupId,
+				CanLeave:        result.CanLeave,
+				AutoAdd:         result.AutoAdd,
+				CreateAt:        result.CreateAt,
+				DeleteAt:        result.DeleteAt,
+				UpdateAt:        result.UpdateAt,
+				Type:            syncableType,
+				TeamDisplayName: result.TeamDisplayName,
+				TeamType:        result.TeamType,
+			}
+			groupSyncables = append(groupSyncables, groupSyncable)
+		}
+	case model.GroupSyncableTypeChannel:
+		sqlQuery := `
+			SELECT
+				GroupChannels.*, 
+				Channels.DisplayName AS ChannelDisplayName, 
+				Teams.DisplayName AS TeamDisplayName,
+				Channels.Type As ChannelType,
+				Teams.Type As TeamType,
+				Teams.Id AS TeamId
+			FROM 
+				GroupChannels 
+				JOIN Channels ON Channels.Id = GroupChannels.ChannelId
+				JOIN Teams ON Teams.Id = Channels.TeamId
+			WHERE 
+				GroupId = :GroupId AND GroupChannels.DeleteAt = 0`
+
+		results := []*groupChannelJoin{}
+		_, err := s.GetMaster().Select(&results, sqlQuery, args)
+		if err != nil {
+			result.Err = appErrF(err.Error())
+			return result
+		}
+		for _, result := range results {
+			groupSyncable := &model.GroupSyncable{
+				SyncableId:         result.ChannelId,
+				GroupId:            result.GroupId,
+				CanLeave:           result.CanLeave,
+				AutoAdd:            result.AutoAdd,
+				CreateAt:           result.CreateAt,
+				DeleteAt:           result.DeleteAt,
+				UpdateAt:           result.UpdateAt,
+				Type:               syncableType,
+				ChannelDisplayName: result.ChannelDisplayName,
+				ChannelType:        result.ChannelType,
+				TeamDisplayName:    result.TeamDisplayName,
+				TeamType:           result.TeamType,
+				TeamID:             result.TeamID,
+			}
+			groupSyncables = append(groupSyncables, groupSyncable)
+		}
+	default:
+	}
+
+	result.Data = groupSyncables
+	return result
+}
+
+func (s *SqlSupplier) GroupUpdateGroupSyncable(ctx context.Context, groupSyncable *model.GroupSyncable, hints ...store.LayeredStoreHint) *store.LayeredStoreSupplierResult {
+	result := store.NewSupplierResult()
+
+	retrievedGroupSyncable, err := s.getGroupSyncable(groupSyncable.GroupId, groupSyncable.SyncableId, groupSyncable.Type)
+	if err != nil {
+		if err == sql.ErrNoRows {
```

Comment:
Off topic, but having seen this a few times, it really sucks that we have to specially propagate `sql.ErrNoRows` upwards. It seems like it would better to expect the following semantics from suppliers:
* non-`sql.ErrNoRows` error: `return nil, err`
* `sql.ErrNoRows`: `return nil, nil`
* `nil` erro: `return result, nil`

Then the app layer can trivially handle a "missing" result instead of having to introspect the error code.